packages/@aws-cdk/aws-servicecatalogappregistry/README.md

```diff
@@ -64,14 +65,85 @@ const importedApplication = appreg.Application.fromApplicationArn(
 );
 ```
 
+## Register-Application
+
+If you want to create an Application named `MyRegisteredApplication` in account `123456789012` and region `us-east-1`
+and want to associate all stacks in the `App` scope to `MyRegisteredApplication`, then use as shown in the example below:
+
+```ts
+const app = new App();
+const registeredApp = new appreg.RegisterApplication(app, 'RegisterApplication', {
+    applicationName: 'MyRegisteredApplication',
+    description: 'Testing registered application',
+    stackProps: {
+        stackName: 'MyRegisteredApplicationStack',
+        env: {account: '123456789012', region: 'us-east-1'},
+    },
+});
+```
+
+If you want to re-use an existing Application with ARN: `arn:aws:servicecatalog:us-east-1:123456789012:/applications/applicationId`
+and want to associate all stacks in the `App` scope to your imported application, then use as shown in the example below:
+
+```ts
+const app = new App();
+const registeredApp = new appreg.RegisterApplication(app, 'RegisterApplication', {
+    applicationArnValue: 'arn:aws:servicecatalog:us-east-1:123456789012:/applications/applicationId',
+    stackProps: {
+        stackName: 'MyRegisteredApplicationStack',
+    },
+});
+```
+
+If you are using CDK Pipelines to deploy your application, the application stacks will be inside Stages, and 
+RegisterApplication will not be able to find them. Call `associateStage` on each Stage object before adding it to the 
+Pipeline, as shown in the example below:
+
+```ts
+import * as cdk from "@aws-cdk/core";
+import * as codepipeline from "@aws-cdk/pipelines";
+import * as codecommit from "@aws-cdk/aws-codecommit";
+declare const repo: codecommit.Repository;
+declare const pipeline: codepipeline.CodePipeline;
+declare const beta: cdk.Stage;
+class ApplicationPipelineStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props: ApplicationPipelineStackProps) {
+    super(scope, id, props);
+    
+   //associate the stage to register application.
+   props.application.associateStage(beta);
+   pipeline.addStage(beta);
+  }
+};
+
+interface ApplicationPipelineStackProps extends cdk.StackProps {
+  application: appreg.RegisterApplication;
+};
+
+const app = new App();
+const registeredApp = new appreg.RegisterApplication(app, 'RegisterApplication', {
```

Comment:
I also don't particularly like this new name. It should be a noun.

`ApplicationRegistrar` then?

Reply:
`RegistryApplication` ?